client/utils/activity/render.ts

```diff
@@ -9,6 +9,13 @@ export const renderActivityItem = (
 ): RenderedActivityItem => {
 	// This cast avoids a TypeScript error:
 	// Expression produces a union type that is too complex to represent. ts(2590)
+	const keyIsPresent = item.kind in activityItemRenderers;
+	if (keyIsPresent === false) {
+		return null as unknown as RenderedActivityItem;
+	}
 	const renderer = activityItemRenderers[item.kind] as ActivityItemRenderer<any>;
-	return renderer(item, context) as RenderedActivityItem;
+
+	return renderer(item, context).message !== null
```

Comment:
See my comments below about null error messages, but also — please prefer storing the result of a call like `renderer(item, context)` in a variable rather than calling it twice to retrieve the same value.